lib/invoker.js

```diff
@@ -24,21 +25,63 @@ const invoke = (userFunction) => {
     };
 };
 
-module.exports = (userFunctionUri, port) => {
-    const userFunction = lookUpFunction(userFunctionUri);
+const registerShutdown = (destroyHook, timeout, server, shutdownSignal) => {
+    return shutdownSignal.then(() => {
+        logger('Graceful shutdown started');
+        return new Promise(((resolve, reject) => {
+            server.tryShutdown(() => {
+                executeDestroy(destroyHook, timeout)
+                    .then(() => {
+                        logger('Graceful shutdown completed');
+                        resolve();
+                    }, (err) => {
+                        logger(`Error during $destroy hook execution: ${err.toString()}`);
+                        reject(err);
```

Comment:
while production code does not need the rejection (and is forced to ignore the rejection - `see server.js`), this makes testing easier

Reply:
propagating the error is important so that the process exits with a meaningful code.

Reply:
yep, now fixed